### PR TITLE
New feature: Deals damage or fires a weapon when crushed

### PR DESCRIPTION
**Pull request by Aephiex**
Local tests:

1. Give civilian infantries with the following codes, and crush them with a Lasher Tank. The Ivan Bomb is confirmed to explode correctly, and even units standing aside can be damaged due to the `CellSpread` defined in the warhead.
- `WhenCrushed.Damage=420`
- `WhenCrushed.Warhead=IvanWH`
- `WhenCrushed.Warhead.Full=true`

2. The same as case 1, except `WhenCrushed.Warhead.Full=false`. Doesn't seem to be any different with case 1.

3. Give civilian infantries with the following codes, and crush them with Yuri's MCV. The vehicle doesn't appear to have been properly attached with the ivan bomb, or I wasn't patient enough to wait for the ticking sound or explosion. Note that, in my client this weapon is modified, which uses Ares custom ivan bomb mechanic instead of the YR mechanic.
- `WhenCrushed.Weapon=IvanBomber`

4. Give civilian infantries with the following codes, and crush them with Yuri's MCV. The death weapon of terrorist explodes and damages the MCV.
- `WhenCrushed.Weapon=TerrorBomb`

### CODE REVIEW

File: src/Ext/TechnoType/Body.h [hunk 1]
```diff
@@ -182,6 +182,17 @@ class TechnoTypeExt
 		Valueable<double> CrushOverlayExtraForwardTilt;
 		Valueable<double> CrushSlowdownMultiplier;
 
+		Nullable<WarheadTypeClass*> WhenCrushed_Warhead;
```
[the remark this answers is not in the document]
Comment (Aephiex) [reply src/Ext/TechnoType/Body.h:185]: Damn, didn't know such a convenience.

**Comment by Aephiex** [comment]
> Ares already has the CrushDamage= and CrushDamage.Warhead= tags, won't this one cause conflicts with it?
> 
> If this is meant to be a reimplementation, i suggest changing tag names from WhenCrushed to CrushDamage and making the Phobos version completely override the Ares version (if it doesn't already). Phobos' animation damage/weapons already does it like that.

I already tested on that case.
1. `CrushDamage=` triggers as normal when alone. No animation will be played, only damage will be dealt.
2. `WhenCrushed.Damage=` triggers as normal when alone. The `Super` warhead animation will be played.
3. If a unit has both tags, both trigger separately as normal.

In other words, the two features work simulteniously and do not have any conflicts.

**Comment by mevitar** [comment]
Why did you close this? I didn't think there was anything wrong with this feature, just that it should disable CrushDamage= due to doing a similar thing.
(but other people might disagree, since Phobos' version of AttachEffect does not disable the original one, and they can work together)

> 3. If I prevent the Ares feature to function to avoid double damage issue, units already configured with `CrushDamage=` will work slightly different than before, as they now cause an explosion when crushed, unlike before they do not explode at all.

So it explodes even when weapon isn't specified? In that case i don't think you need to give it two tags, one for the warhead and one for the weapon. Just the weapon's tag should be enough.

**Comment by Aephiex** [comment]
That's because of my stupidiy I corrupted some of my other pull requests, and I'm so tired I don't want to separate my local changes into multiple pull requests again, so it is now an all-in-one pull request #1438 .

**Comment by mevitar** [comment]
> That's because of my stupidiy I corrupted some of my other pull requests, and I'm so tired I don't want to separate my local changes into multiple pull requests again, so it is now an all-in-one pull request #1438 .

I can't really test that or the other build, no dll were generated. :P
So I tested this one, didn't notice any problems.

**Comment by Aephiex** [comment]
> I can't really test that or the other build, no dll were generated. :P So I tested this one, didn't notice any problems.

No idea how a nightly build is generated.

**Pull request closed without merging**